llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -163,18 +163,25 @@ static bool sinkScalarOperands(VPlan &Plan) {
       if (UI->getParent() == SinkTo)
         return true;
       NeedsDuplicating = UI->onlyFirstLaneUsed(SinkCandidate);
-      // We only know how to duplicate VPRecipeRecipes for now.
-      return NeedsDuplicating && isa<VPReplicateRecipe>(SinkCandidate);
+      // We only know how to duplicate VPRecipeRecipes and
+      // VPScalarIVStepsRecipes for now.
```

Comment:
Wonder if VPScalarIVStepsRecipes should be subject to further optimization after cloning?

Reply:
Are there any optimizations in particular you are thinking about?

Reply:
Optimized considering only first lane is used, as in turning replicate recipe to be uniform.

Reply:
Done, thanks!